crates/uv-git/src/git.rs

```diff
@@ -206,266 +242,151 @@ impl GitRemote {
         paths::create_dir_all(dirname)?;
         if into.exists() {
             paths::remove_dir_all(into)?;
         }
 
-        // we're doing a local filesystem-to-filesystem clone so there should
-        // be no need to respect global configuration options, so pass in
-        // an empty instance of `git2::Config` below.
-        let git_config = git2::Config::new()?;
-
-        // Clone the repository, but make sure we use the "local" option in
-        // libgit2 which will attempt to use hardlinks to set up the database.
-        // This should speed up the clone operation quite a bit if it works.
-        //
-        // Note that we still use the same fetch options because while we don't
-        // need authentication information we may want progress bars and such.
-        let url = Url::from_file_path(&database.path)
-            .map_err(|()| anyhow::format_err!("Invalid path URL: {}", database.path.display()))?;
-        let mut repo = None;
-        with_fetch_options(&git_config, url.as_str(), &mut |fopts| {
-            let mut checkout = git2::build::CheckoutBuilder::new();
-            checkout.dry_run(); // we'll do this below during a `reset`
-
-            let r = git2::build::RepoBuilder::new()
-                // use hard links and/or copy the database, we're doing a
-                // filesystem clone so this'll speed things up quite a bit.
-                .clone_local(git2::build::CloneLocal::Local)
-                .with_checkout(checkout)
-                .fetch_options(fopts)
-                .clone(url.as_str(), into)?;
-            // `git2` doesn't seem to handle shallow repos correctly when doing
-            // a local clone. Fortunately all that's needed is the copy of the
-            // one file that defines the shallow boundary, the commits which
-            // have their parents omitted as part of the shallow clone.
-            //
-            // TODO(git2): remove this when git2 supports shallow clone correctly
-            if database.repo.is_shallow() {
-                fs_err::copy(
-                    database.repo.path().join("shallow"),
-                    r.path().join("shallow"),
-                )?;
-            }
-            repo = Some(r);
-            Ok(())
-        })?;
-        let repo = repo.unwrap();
-
-        let checkout = GitCheckout::new(database, revision, repo);
+        // Perform a local clone of the repository, which will attempt to use
+        // hardlinks to set up the repository. This should speed up the clone operation
+        // quite a bit if it works.
+        ProcessBuilder::new("git")
+            .arg("clone")
+            .arg("--local")
+            // Make sure to pass the local file path and not a file://... url. If given a url,
+            // Git treats the repository as a remote origin and gets confused because we don't
+            // have a HEAD checked out.
+            .arg(database.repo.path.portable_display().to_string())
```

Comment:
I don't know if we should use `portable_display` here -- it will use a relative path in some cases. Should we use `simplified_display`, to get an absolute path?